source/grid/grid_out.cc

```diff
@@ -2943,23 +3013,142 @@ GridOut::write_vtk(const Triangulation<dim, spacedim> &tria,
 {
   AssertThrow(out, ExcIO());
 
-  // convert the cells of the triangulation into a set of patches
-  // and then have them output. since there is no data attached to
-  // the geometry, we also do not have to provide any names, identifying
-  // information, etc.
-  std::vector<DataOutBase::Patch<dim, spacedim>> patches;
-  patches.reserve(tria.n_active_cells());
-  generate_triangulation_patches(patches, tria.begin_active(), tria.end());
-  DataOutBase::write_vtk(
-    patches,
-    triangulation_patch_data_names(),
-    std::vector<
-      std::tuple<unsigned int,
-                 unsigned int,
-                 std::string,
-                 DataComponentInterpretation::DataComponentInterpretation>>(),
-    vtk_flags,
-    out);
+  // get the positions of the vertices
+  const std::vector<Point<spacedim>> &vertices = tria.get_vertices();
+
+  const auto n_vertices = vertices.size();
+
+  out << "# vtk DataFile Version 3.0" << std::endl
+      << "Triangulation generated with deal.II" << std::endl
```

Comment:
I'm just using VTK conventions here. VTK does not allow comments in its "legacy files", so the first `#` is really not a comment, but it is part of the format that VTK expects.

See here, page 2:

https://www.vtk.org/wp-content/uploads/2015/04/file-formats.pdf

Since the entire second line is used as a title for the data (and vtk can output it if asked to), I did not want to have a `#` in front of it...

Reply:
Fair enough. These legacy file formats are always awkward to read because stuff depends on *position* (like "the second line is a comment") and not based on logical markup (as in XML).

Let's not worry about it then.